x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -198,20 +196,10 @@ public PlanFactory visitDissectCommand(EsqlBaseParser.DissectCommandContext ctx)
 
             try {
                 DissectParser parser = new DissectParser(pattern, appendSeparator);
-                Set<String> referenceKeys = parser.referenceKeys();
-                if (referenceKeys.size() > 0) {
-                    throw new ParsingException(
-                        src,
-                        "Reference keys not supported in dissect patterns: [%{*{}}]",
-                        referenceKeys.iterator().next()
-                    );
-                }
-                List<Attribute> keys = new ArrayList<>();
-                for (var x : parser.outputKeys()) {
-                    if (x.isEmpty() == false) {
-                        keys.add(new ReferenceAttribute(src, x, DataType.KEYWORD));
-                    }
-                }
+
+                Dissect.Parser esqlDissectParser = new Dissect.Parser(pattern, appendSeparator, parser);
```

Comment:
Is this on purpose...?
My feeling is that you could safely reuse this object in the `new Dissect()` below, but maybe there is something I'm missing?

Reply:
Good catch, thank you!